lib/trino-plugin-toolkit/src/test/java/io/trino/plugin/base/security/TestRestBasedAccessControl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import io.airlift.bootstrap.Bootstrap;
+import io.trino.plugin.base.CatalogNameModule;
+import io.trino.plugin.base.util.TestHttpServer;
+import io.trino.spi.connector.ConnectorAccessControl;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+
+import java.io.IOException;
+import java.nio.file.Path;
+
+import static io.trino.plugin.base.util.Closables.closeAllSuppress;
+import static java.util.Objects.requireNonNull;
+
+public class TestRestBasedAccessControl
+        extends BaseRuleBasedConnectorAccessControlTest
+{
+    private TestHttpServer testHttpServer;
+
+    @BeforeClass
+    public void setUp()
+    {
+        testHttpServer = new TestHttpServer();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown() throws IOException
+    {
+        closeAllSuppress(new RuntimeException("Error while closing resources"), testHttpServer);
+    }
+
+    @Override
+    protected ConnectorAccessControl createAccessControl(String fileName)
+    {
+        Path configFilePath = Path.of(getResourcePath(fileName));
+        String dataUrl = testHttpServer.resolve(configFilePath.toString()).toString();
+
+        Bootstrap bootstrap = new Bootstrap(new CatalogNameModule("test_catalog"), new RestBasedAccessControlModule());
+
+        Injector injector = bootstrap
+                .doNotInitializeLogging()
+                .setRequiredConfigurationProperties(ImmutableMap.of("security.config-url", dataUrl))
+                .initialize();
+
+        return injector.getInstance(ConnectorAccessControl.class);
+    }
+
+    @Override
+    protected String getErrorMessagePrefix()
+    {
+        return "Failed to convert JSON tree node";
+    }
+
+    protected String getResourcePath(String resourceName)
```

Comment:
this one should be common to both the tests right ?

Reply:
moved to base class